posthog/api/property_definition.py

```diff
@@ -56,42 +58,66 @@ class PropertyDefinitionViewSet(
-                    WHERE team_id = %(team_id)s AND name NOT IN %(excluded_properties)s {name_filter} {search_query}
-                    ORDER BY query_usage_30_day DESC NULLS LAST, name ASC
-                    """,
-                    params={
-                        "names": names,
-                        "team_id": self.team_id,
-                        "excluded_properties": tuple(HIDDEN_PROPERTY_DEFINITIONS),
-                        **search_kwargs,
-                    },
-                )
-                return ee_property_definitions
+                use_entreprise_taxonomy = False
+
+        properties_to_filter = self.request.GET.get("properties", None)
+        if properties_to_filter:
+            names = tuple(properties_to_filter.split(","))
+            name_filter = "AND name IN %(names)s"
+        else:
+            names = ()
+            name_filter = ""
+
+        # Passed as JSON instead of duplicate properties like event_names[] to work with frontend's combineUrl
+        event_names = self.request.GET.get("event_names", None)
+        if event_names:
+            event_names = json.loads(event_names)
+
+        if event_names and len(event_names) > 0:
+            event_property_field = "(SELECT count(1) > 0 FROM posthog_eventproperty WHERE posthog_eventproperty.team_id=posthog_propertydefinition.team_id AND posthog_eventproperty.event IN %(event_names)s AND posthog_eventproperty.property = posthog_propertydefinition.name)"
+        else:
+            event_property_field = "true"
+
+        search = self.request.GET.get("search", None)
+        search_query, search_kwargs = term_search_filter_sql(self.search_fields, search)
+
+        params = {
+            "event_names": tuple(event_names or []),
+            "names": names,
+            "team_id": self.team_id,
+            "excluded_properties": tuple(HIDDEN_PROPERTY_DEFINITIONS),
+            **search_kwargs,
+        }
 
-        return self.filter_queryset_by_parents_lookups(
-            PropertyDefinition.objects.exclude(name__in=HIDDEN_PROPERTY_DEFINITIONS)
-        ).order_by(self.ordering)
+        if use_entreprise_taxonomy:
+            return EnterprisePropertyDefinition.objects.raw(
+                f"""
+                SELECT posthog_propertydefinition.*,
+                       ee_enterprisepropertydefinition.*, 
+                       {event_property_field} AS is_event_property
+                FROM posthog_propertydefinition
+                LEFT JOIN ee_enterprisepropertydefinition ON ee_enterprisepropertydefinition.propertydefinition_ptr_id=posthog_propertydefinition.id
+                WHERE posthog_propertydefinition.team_id = %(team_id)s AND name NOT IN %(excluded_properties)s {name_filter} {search_query}
+                GROUP BY posthog_propertydefinition.id, ee_enterprisepropertydefinition.propertydefinition_ptr_id
+                ORDER BY is_event_property DESC, name ASC
```

Comment:
Non-blocking. Curious why we got rid of the `query_usage_30_day` order param, seems like it could be useful if performance allows it.

Reply:
It's back